src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class NativeIntegerTypeSymbol : WrappedNamedTypeSymbol
+    {
+        internal NativeIntegerTypeSymbol(NamedTypeSymbol underlying) : base(underlying)
+        {
+            Debug.Assert(!underlying.IsNativeInt);
+            Debug.Assert(underlying.SpecialType == SpecialType.System_IntPtr || underlying.SpecialType == SpecialType.System_UIntPtr);
+            Debug.Assert(this.Equals(underlying));
+            Debug.Assert(underlying.Equals(this));
+        }
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override NamedTypeSymbol ConstructedFrom => _underlyingType.ConstructedFrom;
```

Comment:
>_underlyingType.ConstructedFrom [](start = 59, length = 31)

I would expect that it would return itself. Why delegating to the underlying is the right thing to do?